opendevin/runtime/browser/browser_env.py

```diff
@@ -91,13 +91,22 @@ def step(self, action_str: str, timeout: float = 10) -> dict:
                     return obs
 
     def close(self):
+        if not self.process.is_alive():
+            logger.info('BrowserEnv already closed, no need to close again')
+            return
         try:
             self.agent_side.send(('SHUTDOWN', None))
-            self.process.join()
+            self.process.join(5)  # Wait for the process to terminate
+            if self.process.is_alive():
+                logger.error(
+                    'Browser process did not terminate, forcefully terminating...'
+                )
+                self.process.terminate()
+                self.process.join(5)  # Wait for the process to terminate
```

Comment:
@li-boxuan  wondering whether we should `self.process.kill()` if still not terminate?

Reply:
Make sense. I don't think that's necessary here in this PR - I tried many times after the fix and the hung issue was gone. I'll push a change to https://github.com/OpenDevin/OpenDevin/pull/1933 though.